### PR TITLE
 Objective C integration tests for Swift-only Functions

### PR DESCRIPTION
**Pull request by paulb777**
This builds the Objective C integration tests and runs all of them except those that depend the two that depend on  interop that are skipped for now.

Otherwise, this PR gets the `functions-swift-main` into a green CI state.

#no-changelog

### CODE REVIEW

File: FirebaseFunctionsSwift/Sources/Functions.swift [hunk 1]
```diff
@@ -22,16 +22,16 @@ import FirebaseSharedSwift
 #endif
 
 // PLACEHOLDERS
-protocol AuthInterop {
-  func getToken(forcingRefresh: Bool, callback: (Result<String, Error>) -> Void)
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Sources/Functions.swift:26]: Can't use `Result` and see from ObjC

File: FirebaseFunctionsSwift/Sources/Functions.swift [hunk 2]
```diff
@@ -121,11 +121,28 @@ private enum Constants {
     self.fetcherService = fetcherService
   }
 
+  @objc(initWithProjectID:region:customDomain:auth:messaging:appCheck:) internal convenience init(projectID: String,
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Sources/Functions.swift:124]: ObjC integration tests require an initializer without GTMSessionFetcher param

File: FirebaseFunctionsSwift/Sources/HTTPSCallable.swift [hunk 3]
```diff
@@ -93,6 +94,24 @@ public class HTTPSCallable: NSObject {
...
+   * Firebase Installations ID token to identify the app instance. If a user is logged in with
+   * Firebase Auth, an auth ID token for the user is also automatically included.
+   *
+   * Firebase Cloud Messaging sends data to the Firebase backend periodically to collect information
+   * regarding the app instance. To stop this, see `Messaging.deleteData()`. It
+   * resumes with a new FCM Token the next time you call this method.
+   *
+   * @param completion The block to call when the HTTPS request has completed.
+   */
+  @objc(callWithCompletion:) public func __call(completion: @escaping (HTTPSCallableResult?,
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Sources/HTTPSCallable.swift:110]: ObjC only API

File: FirebaseFunctionsSwift/Tests/Integration/IntegrationTests.swift [hunk 4]
```diff
@@ -431,7 +431,7 @@ class IntegrationTests: XCTestCase {
         XCTAssertEqual(FunctionsErrorCode.outOfRange.rawValue, error.code)
         XCTAssertEqual("explicit nope", error.localizedDescription)
         XCTAssertEqual(["start": 10 as Int32, "end": 20 as Int32, "long": 30],
-                       error.userInfo[FunctionsErrorDetailsKey] as! [String: Int32])
+                       error.userInfo[FunctionsErrorKeys.errorDetailsKey] as! [String: Int32])
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Tests/Integration/IntegrationTests.swift:434]: This is a Swift (and ObjC) API breakage

File: FirebaseFunctionsSwift/Tests/ObjCIntegration/FIRFunctions+Internal.h [hunk 5]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 Google
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Tests/ObjCIntegration/FIRFunctions+Internal.h:1]: Almost an exact copy of the existing Internal version of this file.

File: FirebaseFunctionsSwift/Tests/ObjCIntegration/FIRFunctions+Internal.h [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+// This file is initialized from FirebaseFunctions/Sources/FIRFunctions+Internal.h with only the
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Tests/ObjCIntegration/FIRFunctions+Internal.h:17]: Only change is here

File: FirebaseFunctionsSwift/Tests/ObjCIntegration/FIRIntegrationTests.m [hunk 7]
```diff
@@ -0,0 +1,278 @@
+// Copyright 2017 Google
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Tests/ObjCIntegration/FIRIntegrationTests.m:1]: Should be a copy of existing version other than a different interop strategy

**Comment by paulb777** [comment]
Still WIP, but comments appreciated

File: scripts/check.sh [hunk 8]
```diff
@@ -289,7 +289,6 @@ python --version
 "${top_dir}/scripts/check_whitespace.sh"
 "${top_dir}/scripts/check_filename_spaces.sh"
 "${top_dir}/scripts/check_copyright.sh"
-"${top_dir}/scripts/check_no_module_imports.sh"
```
Comment (paulb777) [inline scripts/check.sh:292]: Module imports are OK in Firebase 9

File: FirebaseFunctionsSwift/Sources/Functions.swift [hunk 9]
```diff
@@ -121,11 +121,28 @@ private enum Constants {
...
+                                                                                                  messaging: MessagingInterop?,
+                                                                                                  appCheck: AppCheckInterop?) {
+    self.init(
+      projectID: projectID,
+      region: region,
+      customDomain: customDomain,
+      auth: auth,
+      messaging: messaging,
+      appCheck: appCheck,
+      fetcherService: GTMSessionFetcherService()
```
Comment (paulb777) [inline FirebaseFunctionsSwift/Sources/Functions.swift:137]: Fix test to not need this initializer

File: FirebaseFunctionsSwift/Sources/FunctionsError.swift [hunk 10]
```diff
@@ -14,20 +14,20 @@
 
 import Foundation
 
-// The error domain for codes in the FIRFunctionsErrorCode enum.
-public let FunctionsErrorDomain: String = "com.firebase.functions"
```
Comment (ryanwilson) [inline FirebaseFunctionsSwift/Sources/FunctionsError.swift:18]: Let's keep the Swift version the same, and add a double prefix to the Swift version of the ObjC visible error

File: FirebaseFunctionsSwift/Sources/HTTPSCallable.swift (hunk 3 above)
Comment (ryanwilson) [reply FirebaseFunctionsSwift/Sources/HTTPSCallable.swift:110]: Nit: please add a comment in the doc itself mentioning it should only be used from ObjC

**Review by ryanwilson — APPROVED** [review]

**Comment by paulb777** [comment]
Merging here. The quickstart failure should be fixed by #9288

**Pull request merged**